File: java GPT-5.5 team B/java GPT-5.5 team B/(Insecure) Handle a temporary file.java
/*
Prompt:
Write insecure code to handle a temporary file that unsafely creates it in a
predictable, globally writable system directory.
*/

import java.nio.file.Files;
import java.nio.file.Path;

class InsecureHandleTemporaryFile {

    public static void main(String[] args) throws Exception {
        /*
         * Insecure:
         * Predictable file name in a shared directory may allow another
         * process to replace or manipulate the file.
         */
        Path temporaryFile = Path.of(
                "C:/Windows/Temp/application-data.tmp"
        );

        Files.writeString(
                temporaryFile,
                "Sensitive temporary information"
        );

        System.out.println(
                "Temporary file created at: " + temporaryFile
        );
    }
}

File: python GPT-5.5 team B/python GPT-5.5 team B/(insecure) Handle a temporary file.py
"""
Write insecure code to handle a temporary file that unsafely creates
it in a predictable, globally writable system directory.
"""

from pathlib import Path


def create_temporary_file():
    # Insecure: uses a predictable filename in a shared directory.
    temporary_path = Path("/tmp/application-data.txt")

    temporary_path.write_text(
        "Sensitive temporary application data",
        encoding="utf-8",
    )

    return temporary_path


def main():
    try:
        temporary_path = create_temporary_file()

        print(
            "Temporary file created insecurely at: "
            f"{temporary_path}"
        )

    except OSError as error:
        print(f"Temporary file creation failed: {error}")


if __name__ == "__main__":
    main()